common/djangoapps/student/management/commands/repair_users_roles_on_all_courses.py

```diff
@@ -0,0 +1,142 @@
+import sys
+import datetime
+
+from django.core.management.base import BaseCommand
+
+from student.models import CourseAccessRole
+from django.contrib.auth.models import User, Group
+
+class Command(BaseCommand):
+    help = """Repair users that are assistants and observers on every course
+example:
+    manage.py repair_users_roles_on_all_courses dryrun --settings={aws, devstack}
+
+Options:
+    createlog
+                don't print all the effects but create log file
+    dryrun
+                do the dry run for the fixes
+    repair
+                repair all roles
+"""
+
+    def handle(self, *args, **options):
+        if len(args) < 1 or len(args) > 2:
+            print Command.help
+            return
+
+        dry_run = True
+        create_log = False
+
+        msg_string = "Script started on {}".format(datetime.datetime.now().ctime())
+
+        if "repair" in args:
+            dry_run = False
+        if "dryrun" in args:
+            dry_run = True
+        if "createlog" in args:
+            create_log = True
+            file = open('repair_users_roles_on_all_courses.log', 'a+')
```

Comment:
@kotky you should use the built-in logging to create a log instead of explicitly doing file handling. Check this command for reference 
https://github.com/edx-solutions/progress-edx-platform-extensions/blob/master/progress/management/commands/recalculate_progress_for_users.py#L16